pkg/cleaner/cleaner.go

```diff
@@ -24,12 +24,14 @@ func Run(gitHome string, nsLister k8s.NamespaceWatcher, fs sys.FS) error {
 	watcher, err := nsLister.Watch(nil, nil, "")
 	if err != nil {
 		log.Printf("unable to get watch events (%s)", err)
+		return err
```

Comment:
there is code in boot.go(https://github.com/deis/builder/blob/master/boot.go#L110-L112) which is handling the errors and restarting the container but we are never returning the error.

The below code is to restart the process if the channel has closed.